sdk/java/logbat-sdk/src/main/java/info/logbat/infrastructure/LogProcessScheduler.java

```diff
@@ -0,0 +1,121 @@
+package info.logbat.infrastructure;
+
+import info.logbat.infrastructure.payload.LogSendRequest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+public class LogProcessScheduler {
+
+    private static final Integer DEFAULT_BULK_SIZE = 100;
+    private static final Integer DEFAULT_INTERVAL = 2000;
+    private static final Integer TERMINATION_TIMEOUT = 60;
+    private static final Integer SHUTDOWN_THREAD_POOL_SIZE = 4; // 병렬 처리를 위한 스레드 풀 크기
+
+    private final Consumer<List<LogSendRequest>> sendFunction;
+    private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private final LogBuffer logBuffer;
+
+
+    /**
+     * LogProcessScheduler는 주기적으로 LogBuffer에 저장된 로그를 읽어와 전송하는 스케줄러입니다. logBuffer에 로그가 없을 경우 아무 동작도
+     * 하지 않습니다.
+     * <p>
+     * Shutdown hook을 등록하여 JVM 종료 시 gracefulShutdown을 호출합니다.
+     *
+     * @param sendFunction
+     * @param logBuffer
+     */
+    public LogProcessScheduler(Consumer<List<LogSendRequest>> sendFunction, LogBuffer logBuffer) {
+        this.sendFunction = sendFunction;
+        this.logBuffer = logBuffer;
+        startScheduling();
+        Runtime.getRuntime().addShutdownHook(new Thread(this::shutdownSafely));
+    }
+
+    private void startScheduling() {
+        scheduler.scheduleAtFixedRate(
+            this::processLogs,
+            0,
+            DEFAULT_INTERVAL,
+            TimeUnit.MILLISECONDS
+        );
+    }
+
+    private void processLogs() {
+        // catch block에서 logs를 잡아 retryQueue에 넣는 로직을 구현하기 위해 try catch block 이전에 선언했습니다.
+        List<LogSendRequest> logs = new ArrayList<>();
+        try {
+            logs.addAll(logBuffer.getLogs(DEFAULT_BULK_SIZE));
```

Comment:
logBuffer로 부터 받은 로그들을 try 블럭 안에서만 소진하는 생각이 들어서 아래처럼 하지 않으신 이유가 있을까요?
혹시 실패할 경우 `logs` 를 catch 블러 안에서 처리하려고 밖에 빼두신 걸까요?
그렇다면 ` = new ArrayList<>();` 까지는 필요하지 않을 것 같습니다! 
```suggestion
        try {
            List<LogSendRequest> logs = logBuffer.getLogs(DEFAULT_BULK_SIZE);
```

Reply:
아! 이 부분을 실패 시 catch 블럭에서 retryQueue까지 넣는 로직으로 구현 계획했으나 이번 PR에서는 제외시킨 부분이라 위와 같이 구현이 되어 있습니다!!